### PR TITLE
Document codec server protocol.

### PR DESCRIPTION
**Pull request by robholland**
## What was changed
Added a document outlining codec server protocol.

## Why?
To help users understand how the protocol works and developers produce implementations suitable for their use cases.

### CODE REVIEW

File: codec-server/protocol.md [hunk 1]
```diff
@@ -0,0 +1,66 @@
+# Codec Server: protocol
```
Comment (cretz) [inline codec-server/protocol.md:1]: I think these details can go right in the README, but if you don't want to, may at least want to link from there.

Comment (robholland) [reply codec-server/protocol.md]: sg, moved.

**Review by cretz — APPROVED** [review]

File: codec-server/README.md [hunk 2]
```diff
@@ -27,3 +27,70 @@ go run ./codec-server
...
+sequenceDiagram;
+	participant tctl
+	participant Server as Codec Server
+
+	tctl->>Server: HTTP POST /encode
+	Note right of tctl: Content-Type: application/json
+	Note right of tctl: Body: Payloads protobuf as JSON
+	alt invalid JSON
+		Server-->>tctl: HTTP 400 BadRequest
+    else decoder error
```
Comment (Sushisource) [inline codec-server/README.md:67]: Should be encoder error?

Comment (robholland) [reply codec-server/README.md:67]: Thanks!

**Pull request merged**